doc/arch/adr-001-notify-integration.md

```diff
@@ -0,0 +1,424 @@
+- Have the ability to be disabled if there are problems with a users account
+- Store other personalisation information such as name
+- Be associated with statistics regarding the subscriber
+
+#### To determine
+- What should we allow in terms of removing a subscriber - do we use the
+  associated data as a log?
+
+### Subscription
+
+**Responsibility**: Store that a Subscriber has an association to a SubscriberList
+and any associated data (e.g. preferences)
+
+#### Shall
+
+- Store an association to a Subscriber
+- Store an association to a Subscribable
+
+#### Should
+
+- Be used to store preferences associated with a particular subscription
+
+#### Should Not
+
+- Be removed from the system when a user unsubscribes from a Subscribable as
+  this provides a log that a user has been subscribed with the associated
+  content and maintaining the record allows us to restore preferences if
+  subscription is resumed
+
+#### May
+
+- Be used to store statistics on a subscribers activity relating to that
+  subscription
+- In future be something that can be paused
+
+#### To determine
+
+- What should happen if either Subscribable or Subscriber are removed/archived
+
+### SubscriptionContent
+
+**Responsibility**: Represent that a content change should or has been emailed to a
+subscription
+
+#### Shall
+
+- Have an association with a ContentChange
+- Have an association with a Subscription
+- Have a nullable association with Email
+- Know whether it has been processed
+
+#### Should
+
+- Allow for multiple subscriptions associated with the same Subscriber to be
+  associated with the same ContentChange - It is the EmailGeneration
+  responsibility to consolidate these
+- Allow for multiple entries with same ContentChange and Subscription - to
+  allow a content change to be reprocessed in the event of an error
+- Support Email records being removed from the system for auditing process
+- Be deleteable for a scenario that a user changes a subscription before an
+  email is sent
+
+#### To determine
+
+- What is the archiving strategy on these? Will they be persisted forever or
+  match persistence strategy of Emails for instance?
+
+### Email
+
+**Responsibility**: A distinct email that the system will be sending or has sent
+
+#### Shall
+
+- Be created any time Email Alert API sends an email
+- Store information such as
+  - Recipient
+  - Subject
+  - Body (in Notify Markdown)
+- Delivery status
+
+#### Shall Not
+
+- Have more than one recipient
+- Know anything about the content that led to its generation
+
+#### Should
+
+- Only exist in the database for a non permanent time period
+- Be associated with a Subscriber for logging purposes
+
+#### May
+
+- Be archived into a log somewhere for later analysis
+
+### DeliveryAttempt
+
+**Responsibility**: Storing the information used to communicate with the
+transportation service for email
+
+#### Shall
+
+- Store information such as
+  - Email associated with
+  - Provider (e.g. Notify)
```

Comment:
Is this definitely required or are we predicting a future where we officially support different providers?